utils/src/buffer.rs

```diff
@@ -0,0 +1,54 @@
+pub trait Writable {
+    fn write(&mut self, bytes: &[u8]);
+}
+
+pub struct Buffer<'a, W> {
+    index: u8,
+    data: &'a [u8],
+    temp_data: *mut W,
+}
+
+impl<'a, W> Buffer<'a, W> {
+    pub fn new(data: &'a [u8], temp_data: *mut W) -> Option<Buffer<'a, W>> {
+        if data.len() > (u8::MAX as usize) {
```

Comment:
Because in ledger APDU protocol, the max message payload is 256 bytes, if the total data length exceeds 256 bytes, it needs to split into multiple frames to send the data. So using `u8::Max` here is enough.